### ISSUE TITLE
Quick Settings tile should not be able to enable server without unlocking device

### ISSUE THREAD

**Issue by kaoneko** [ppareit/swiftp#214]
As it is now, when you add the QS tile, anyone who gets a hold of your phone can turn on the FTP server and access your internal storage. Or whatever directories you're sharing. I think people will not be aware of this when adding the tile (I wasn't at first).

**Comment by Xavron** [comment]
Just don't use anonymous or a bad password (which should be a given for security anyway).

The password isn't visible except in the app which does require unlocking to get to.

Although the password is dumped in this version to logcat, that requires first connecting with the password for that to happen, which mutes that point there. Although, not of the password in the logcat as logcat logs can get sent out with bug issues - removing that is of another matter I brought up previously.

The username is also not known or seen anywhere which would be needed for logins.

There is no mechanism to stop automated brute force attacks either.

Also, newer Android versions have a setting to have WIFI disabled until on previously connected SSID and trying for new ones to connect to is also behind unlocking. You can select which SSID is allowed from Android settings at Android Settings > Connectivity > More button > Intelligent WIFI > Turn WIFI on/off automatically left side which brings up a page to selecting.

Just seeing the server IP and port, turning it on, or having it on while the device is locked isn't itself an issue to me. But, there are other security things to think about :)

**Comment by Xavron** [comment]
TLDR: Personally, I don't see anything to do here except in other ways for security. Someone else might have another opinion on it. I won't be doing anything with it but perhaps someone else might.

Just don't use anonymous or a bad password (which should be a given for security anyway). While not having the app run, better would be to remove the password from the app so even if it gets started while the device is locked such as from ADB or another issue, then there's nothing to be used.

**Comment by kaoneko** [comment]
Oops, totally forgot it requires authentication 😅 I'm so used to enabling the FTP Server and connecting to it from my computer that I forgot my computer has the credentials memorized. Not an issue then!

**Comment by Xavron** [comment]
It could still be a good idea anyway. Idk.

Just to me there's other ways of looking at it like even, newer Android versions have a setting to disable wifi and enable it when back at specific SSID's of which I use. So, that mostly voids the issue as well.

I don't have an issue with leaving the app enabled but that's me and I have too many other thoughts on it and security.

Someone might want to anyway :)

**Comment by Xavron** [comment]
I now expect to look at this in some months (at least 1 month or longer) and create a pull request if no one else does it by then. It should only take a quick moment. Previously, I wasn't going to do anything further but things changed =)

**Comment by Xavron** [comment]
Pull request added https://github.com/ppareit/swiftp/pull/226

Build from source can use that pull request right now. Otherwise, need to wait for ppareit to accept it and then put out app builds with it included.

Tested as working on Android 6 (current app minimum), Android 7.1.1, and Android 14.

**Comment by ppareit** [comment]
Fixed in #226
